src/execution/operator/persistent/physical_delete.cpp

```diff
@@ -35,36 +36,75 @@ class DeleteGlobalState : public GlobalSinkState {
 class DeleteLocalState : public LocalSinkState {
 public:
 	DeleteLocalState(ClientContext &context, TableCatalogEntry &table,
-	                 const vector<unique_ptr<BoundConstraint>> &bound_constraints) {
+	                 const vector<unique_ptr<BoundConstraint>> &bound_constraints)
+	    : has_unique_indexes(false) {
+
 		delete_chunk.Initialize(Allocator::Get(context), table.GetTypes());
-		delete_state = table.GetStorage().InitializeDelete(table, context, bound_constraints);
+		auto &storage = table.GetStorage();
+		delete_state = storage.InitializeDelete(table, context, bound_constraints);
+
+		// We need to append deletes to the local delete-ART.
+		if (storage.HasUniqueIndexes()) {
+			storage.InitializeLocalAppend(append_state, table, context, bound_constraints);
```

Comment:
Maybe we can extract the code that initializes the local storage from the append so that we don't need to call InitializeAppend here but can instead call e.g. InitializeLocalStorage

```cpp
void LocalStorage::InitializeLocalStorage(DataTable &table) {
	table.InitializeIndexes(context);
	state.storage = &table_manager.GetOrCreateStorage(context, table);
}

```

Reply:
This can also be moved to the global state instead of the local state